init.sh

```diff
@@ -45,23 +45,37 @@ function start_zmq_adaptor()
         zmq_adaptor_cmd="${zmq_adaptor_cmd} -d"
     fi
 
-    sudo $zmq_adaptor_cmd >> /dev/stderr &
-    sleep 3s
+    $zmq_adaptor_cmd >> /dev/stderr &
+    sleep 10s
     # allow interprocess communication by allowing api-gateway processes to write to the socket
-    sudo /bin/chown nginx-api-gateway:nginx-api-gateway /tmp/nginx_queue_listen
+    /bin/chown nginx-api-gateway:nginx-api-gateway /tmp/nginx_queue_listen
 }
 # keep the zmq adaptor running using a simple loop
 while true; do zmq_pid=$(ps aux | grep api-gateway-zmq-adaptor | grep -v grep) || ( echo "Restarting api-gateway-zmq-adaptor" && start_zmq_adaptor ); sleep 60; done &
 
 
 echo "Starting api-gateway ..."
+
+was_empty="false"
+
+while ! [ "$(ls -A /etc/api-gateway/generated-conf.d)" ]; do
```

Comment:
This functionality should be configurable.

Reply:
yes it should be but a known issue is that if the gateway starts and the confs come pooring down the pipe as they are beeing downloaded you will have a massive chunk of reloads.